src/server/verification/verificationAPI.js

```diff
@@ -13,96 +12,87 @@ import { sendOTP, generateOTP } from '../../imports/otp'
-      const sessionId = body.sessionId
-      GunDBPublic.gun
-        .get(sessionId)
-        .get('isStarted')
-        .put(true) // publish initialized data to subscribers
-      log.debug('written FR status to gun', { data: await GunDBPublic.gun.get(sessionId) })
-
-      const verificationData = {
-        facemapFile: _.get(_.find(files, { fieldname: 'facemap' }), 'path', ''),
-        auditTrailImageFile: _.get(_.find(files, { fieldname: 'auditTrailImage' }), 'path', ''),
-        enrollmentIdentifier: body.enrollmentIdentifier,
-        sessionId: sessionId
-      }
-      let result = { ok: 0 }
-      if (!user.isVerified && !conf.skipFaceRecognition)
-        result = await verifier
-          .verifyUser(user, verificationData)
-          .catch(e => {
-            log.error('Facerecognition error:', { e })
-            GunDBPublic.gun.get(sessionId).put({ isNotDuplicate: false, isLive: false, isError: e.message })
-            return { ok: 1, error: e.message, isVerified: false }
-          })
-          .finally(() => {
-            //cleanup
-            log.info('cleaning up facerecognition files')
-            fsPromises.unlink(verificationData.facemapFile)
-            fsPromises.unlink(verificationData.auditTrailImageFile)
-          })
-      else {
-        GunDBPublic.gun.get(sessionId).put({ isNotDuplicate: true, isLive: true, isEnrolled: true }) // publish to subscribers
-        // mocked result for verified user or development mode
-        result = {
-          ok: 1,
-          isVerified: true,
-          enrollResult: { alreadyEnrolled: true, enrollmentIdentifier: verificationData.enrollmentIdentifier }
-        } // skip facereco only in dev mode
-      }
-      log.debug('Facerecogintion result:', result)
-      if (result.isVerified) {
-        log.debug('Whitelisting new user', user)
-        await Promise.all([
-          AdminWallet.whitelistUser(user.gdAddress, user.profilePublickey),
-          storage
-            .updateUser({ identifier: user.loggedInAs, isVerified: true })
-            .then(updatedUser => log.debug('updatedUser:', updatedUser))
-        ])
-          .then(
-            _ =>
-              GunDBPublic.gun
-                .get(sessionId)
-                .get('isWhitelisted')
-                .put(true) // publish to subscribers
-          )
-          .catch(e => {
-            log.error('Whitelisting failed', e)
-            GunDBPublic.gun.get(sessionId).put({ isWhitelisted: false, isError: e.message })
-          })
+    wrapAsync(async (req, res) => {
+      const { user, log, params, body: payload } = req
+      const { sessionId } = payload
+      const { enrollmentIdentifier } = params
+
+      let enrollmentResult
+      const processor = createEnrollmentProcessor(storage)
+
+      try {
+        processor.validate(user, enrollmentIdentifier, payload)
+
+        if (user.isVerified || conf.skipFaceVerification) {
+          const sessionRef = GunDBPublic.session(sessionId)
```

Comment:
if user.isVerified we need to make sure to whitelist him, maybe we changed the whitelisting contract and he is no longer whitelisted there, so we trust that we already whitelistedhim in the past and whitelist him again in the new contract